exasol/analytics/query_handler/query/select.py

```diff
@@ -44,30 +50,41 @@ def audit_fields(self) -> dict[str, Any]:
         return self._audit_fields
 
 
-class AuditQuery(AuditData):
+class AuditQuery(SelectQueryWithColumnDefinition, AuditData):
     """
-    A query wrapper for a special read-only query that selects data for auditing.
+    A wrapper for a special read-only query that selects data for auditing. An object
+    of the class can also be used as an audit property bag, since it inherits from the
+    `AuditData` as well as `SelectQueryWithColumnDefinition`. The query is optional.
+    If provided the output columns should also be defined.
     """
-    def __init__(self, query_with_columns: SelectQueryWithColumnDefinition,
+    def __init__(self, query_string: str = '',
+                 output_columns: list[Column] | None = None,
                  audit_fields: dict[str, Any] | None = None):
         super().__init__(audit_fields)
-        self._query_with_columns = query_with_columns
+        SelectQueryWithColumnDefinition.__init__(query_string, output_columns or [])
+        AuditData.__init__(self, audit_fields)
 
-    def query_with_columns(self) -> SelectQueryWithColumnDefinition:
-        return self._query_with_columns
+    def __post_init__(self):
+        if self.query_string and not self.output_columns:
+            raise RuntimeError('No columns defined for an audit query')
 
 
 class WriteQuery(CustomQuery, AuditData):
     """
-    A query wrapper for a query that changes data in a database table (e.g. INSERT or
-    UPDATE) or creates the table (e.g. CREATE TABLE). This type of query is auditable.
+    A wrapper for a query that changes data in a database table (e.g. INSERT or UPDATE)
+    or creates the table (e.g. CREATE TABLE). This type of query is auditable.
     """
-    def __init__(self, query_string: str, audit: bool = False,
-                 audit_fields: dict[str, Any] | None = None):
+    def __init__(self, query_string: str, affected_table: TableLikeName,
```

Comment:
if we do it this way, we probably should rename the class to WriteTableQuery and add another property operation, which can be of value INSERT, DELETE, CREATE, DROP, TRUNCATE

Reply:
OK, if we also need to record the operation then we have two options - multiple classes or an extra parameter. I think they are equivalent. The latter is probably more compact.
The name `WriteTableQuery` kinda excludes the CREATE TABLE case. By `WriteQuery` I meant writing some to the database, as opposed to selecting the data and leaving the database unchanged.

Reply:
What is db_object_type? Do you still want the INSERT, DELETE, CREATE, DROP, TRUNCATE?

Reply:
db_object_type would be TABLE, SCHEMA, ..., the INSERT, DELETE, CREATE would be the operation which also need. Having the db object name, type and the operation allows us to keep the Write/ModifyQuery very generic

Reply:
Good question, enumeration is probably saner, but maybe do, Enum | str, for the case of a consumer of the AAF has specific requirements